### ISSUE TITLE
VSCode: ignore flag to skip validation on a query

### ISSUE THREAD

**Issue by TdyP** [apollographql/apollo-tooling#1535]
Query validation on VSCode is a very nice feature. Though, sometimes it would be appreciable to disable validation on a specific query.
I was thinking of a flag like all linters have (eg. `/* tslint:disable-next-line */`) or something like that.

My use case:
I have a query where the query name itself is dynamic, so I'm replacing it in my template literal, like that:
```
const queryName = 'someQuery';
const query = gql`
        ${queryName} {
            id
            ...
        }
`;
```
It is totally valid and working at runtime, but the validation keeps complaining about a syntax error.

Thanks!

**Comment by Jonatthu** [comment]
I need this also @trevor-scheer

**Comment by trevor-scheer** [comment]
Hey @TdyP, Apollo generally recommends against this practice. Having queries be statically analyzable brings life to our tooling (Studio, codegen, VS Code, operation registry). Since this query is dynamic and can only be evaluated at runtime, static analysis and all the benefits that come from it is not possible.

If you expect that down the line you might become _more invested_ in the tooling that Apollo offers, I think you'd find it beneficial not to do this.

**Comment by TdyP** [comment]
@trevor-scheer Yes, I'm totally aware of that. I only have a very few dynamic queries. I basically have 2 use cases where it makes sense to me:
- My datamodel is very dynamic, highly customizable by user. Schema is generated based on these customisations, reflecting exactly the user business rules and entities. Thus, I have a query with a dynamic name. I could totally handle this with a generic query and some variables, but I think in most cases the DX is better with these specialized queries
- Fetching recursive data of unknown depth. Graphql is very bad for this. Thus, my front app generate a query for the depth it needs

**Comment by trevor-scheer** [comment]
@TdyP thanks for the additional context! I'm glad you have a workaround. I'm going to *reopen* this issue however, as we've neither provided a proper solution nor decided _against_ supporting something like this. I can't promise we'll be addressing it any time soon but I think it's valuable to keep this open for history and context, nonetheless.

Thanks again for submitting and informing others of the workaround 😄

**Comment by AlexanderWells-diamond** [comment]
I can add another use-case to this problem: When multiple schemas are involved. I haven't found a way to allow multiple schema definitions in the apollo.config.js file, but my application has to call two different GraphQL servers. Hence any queries I define for the second schema are flagged as errors.

So I suppose the actual solution is to provide a mechanism for multiple schemas, but I'd settle for a way to just suppress the errors. Thanks.
